src/prefect/projects/steps/core.py

```diff
@@ -74,4 +75,8 @@ async def run_step(step: dict) -> dict:
     inputs = await resolve_variables(inputs)
 
     step_func = _get_function_for_step(fqn, requires=keywords.get("requires"))
-    return step_func(**inputs)
+    output = step_func(**inputs)
+    if inspect.isawaitable(output):
+        return await output
+    else:
+        return output
```

Comment:
Technically we should probably do something like `from_async.call_soon_in_worker_thread(...)` which would prevent synchronous user code from blocking the event loop and take care of running the async user code in an isolated loop. That can be a future optimization though.

Reply:
Good call! I updated `run_step` to use `from_sync.wait_for_call_in_new_thread` which I think is the right function for this use case, and seem to work, but let me know if there's a different function that would work better.